pkg/agent/openflow/pipeline.go

```diff
@@ -1242,6 +1394,41 @@ func (c *client) l3FwdFlowToRemoteViaGW(
 		Done()
 }
 
+// l3FwdServiceDefaultFlowsViaGW generates the default L3 forward flow to support Service traffic to pass through Antrea gateway.
+func (c *client) l3FwdServiceDefaultFlowsViaGW(ipProto binding.Protocol, category cookie.Category) []binding.Flow {
+	gatewayMAC := c.nodeConfig.GatewayConfig.MAC
+
+	flows := []binding.Flow{
+		/* This flow is used to match the packets of Service traffic:
+			- NodePort/LoadBalancer request packets which passes through Antrea gateway and its Endpoint is on host network.
+			- ClusterIP request packets which are from Antrea gateway and its Endpoint is on host network.
+		  The matched packets should leave through Antrea gateway, however, they also arrives through Antrea gateway. This
+		  is hairpin traffic and it requires SNAT.
```

Comment:
Do you mean the packet will be SNAT'd by other flows (I do not see anything about SNAT in the flows in this func)? Probably make it clear.

Reply:
> Do you mean the packet will be SNAT'd by other flows (I do not see anything about SNAT in the flows in this func)? Probably make it clear.

SNAT will be performed in Table 106. I will delete the comment to avoid misunderstandings.